src/jsrm/systems/planar_pcs_num.py

```diff
@@ -0,0 +1,1280 @@
+        J = jacobian(lambda xi: chi_fn_xi(params, xi, s))(xi)
+
+        # apply the strain basis to the Jacobian
+        J = J @ B_xi
+
+        return J
+
+    @jit
+    def J_explicit(
+        params: Dict[str, Array],
+        xi: Array,
+        s: Array,
+        eps: float = global_eps
+    ) -> Array:
+        """
+        Compute the Jacobian of the forward kinematics function with respect to the strain vector at a given point s.
+
+        Args:
+            params (Dict[str, Array]): dictionary of robot parameters.
+            xi (Array): strain vector of the robot.
+            s (Array): point coordinate along the robot in the interval [0, L_tot].
+            eps (float, optional): small number to avoid singularities. Defaults to global_eps.
+
+        Returns:
+            Array: Jacobian of the forward kinematics function with respect to the strain vector using explicit expression in SE(3).
+        """
+        # Add a small number to the bending strain to avoid singularities
+        xi = apply_eps_to_bend_strains(xi, eps)
+        
+        # Classify the point along the robot to the corresponding segment
+        segment_idx, _, l_cum = classify_segment(params, s)
+        
+        # Indices to extract/inject kappa_z, sigma_x, sigma_y from xi
+        SE2_to_SE3_indices = (2, 3, 4)  
+        
+        # Initial condition
+        xi_SE3_0 = vec_SE2_to_xi_SE3(xi[0:3], SE2_to_SE3_indices)
+        
+        # TODO: check if we better use inv or solve
+        # Ad_g0_inv_L0 = jnp.linalg.inv(Adjoint_gn_SE3(xi_SE3_0, l_cum[0], l_cum[1]))
+        # Ad_g0_inv_s = jnp.linalg.inv(Adjoint_gn_SE3(xi_SE3_0, l_cum[0], s))
+        Ad_g0_inv_L0 = jnp.linalg.solve(Adjoint_gn_SE3(xi_SE3_0, l_cum[0], l_cum[1]), jnp.eye(6))
+        Ad_g0_inv_s = jnp.linalg.solve(Adjoint_gn_SE3(xi_SE3_0, l_cum[0], s), jnp.eye(6))
+        
+        T_g0_L0 = Tangent_gn_SE3(xi_SE3_0, l_cum[0], l_cum[1])
+        T_g0_s = Tangent_gn_SE3(xi_SE3_0, l_cum[0], s)
+
+        mat_0_L0 = Ad_g0_inv_L0 @ T_g0_L0
+        mat_0_s = Ad_g0_inv_s @ T_g0_s
+        
+        J_0_L0 = jnp.zeros((num_segments, 6, 6)).at[0].set(mat_0_L0)
+        J_0_s = jnp.zeros((num_segments, 6, 6)).at[0].set(mat_0_s)
+        
+        tuple_J_0 = (J_0_L0, J_0_s)
+        
+        # Iteration function
+        def J_i(
+            tuple_J_prev: Array,
+            i: int
+        ) -> Array:
+            J_prev_Lprev, _ = tuple_J_prev
+            
+            start_index = 3 * i
+            xi_i = lax.dynamic_slice(xi, (start_index,), (3,))
+            xi_SE3_i = vec_SE2_to_xi_SE3(xi_i, SE2_to_SE3_indices)
+            
+            # TODO: check if we better use inv or solve
+            # Ad_gi_inv_Li = jnp.linalg.inv(Adjoint_gn_SE3(xi_SE3_i, l_cum[i], l_cum[i+1]))
+            # Ad_gi_inv_s = jnp.linalg.inv(Adjoint_gn_SE3(xi_SE3_i, l_cum[i], s))
+            Ad_gi_inv_Li = jnp.linalg.solve(Adjoint_gn_SE3(xi_SE3_i, l_cum[i], l_cum[i+1]), jnp.eye(6))
```

Comment:
Does the Adjoint have some structure such that you can make use of the Schur complement (https://en.wikipedia.org/wiki/Schur_complement)? This should (theoretically) increase the performance.